code/modules/research/xenoarchaeology/artifact/effects/unknown_effect_darkrevive.dm

```diff
@@ -0,0 +1,75 @@
+/datum/artifact_effect/darkrevive
+	effecttype = "darkrevive"
+
+/datum/artifact_effect/darkrevive/New()
+	..()
+	effect = pick(EFFECT_TOUCH)
+	effect_type = pick(0,2,5)
+
+/datum/artifact_effect/darkrevive/DoEffectTouch(var/mob/living/carbon/human/user)
+	if(holder && !isplasmaman(user) && !isskellington(user) && !isgolem(user) && !user.isDead() && user.dna.mutantrace != "shadow")
+		var/list/targets = list()
+		FOR_DVIEW(var/mob/living/carbon/human/H,world.view,get_turf(holder),0)
+			if(!H.mind)
+				continue
+			if(isplasmaman(H) || isskellington(H) || isgolem(H))
+				continue
+			if(H.dna.mutantrace == "shadow")
+				continue
+			var/datum/organ/external/head/head = H.get_organ(LIMB_HEAD)
+			if(!head || head.status & ORGAN_DESTROYED || M_NOCLONE in H.mutations  || !H.has_brain())
+				continue
+			if(H.isDead())
+				targets += H
+
+		if(targets.len)
+			var/mob/living/carbon/human/target = pick(targets)
+			try_revive(user, target)
+
+/datum/artifact_effect/darkrevive/proc/try_revive(var/mob/living/carbon/human/user, var/mob/living/carbon/human/target)
```

Comment:
The ghost bwoink is from defibs, and the reviving is parts of defib and rejuvinate jammed together